From: Stephan Linz <linz@li-pro.net>
Date: Sun, 1 Oct 2017 13:56:02 +0200
Subject: support system wide startup file as last choice

The new search order will be:

  ${PWD}/.yazerc
  ${HOME}/.yazerc
  $(SYSCONFDIR)/yazerc (/usr/local/etc/yazerc)

Signed-off-by: Stephan Linz <linz@li-pro.net>
---
 Makefile | 4 +++-
 bios.c   | 8 ++++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index aa82818..83d609c 100644
--- a/Makefile
+++ b/Makefile
@@ -11,6 +11,7 @@ CC            =	gcc
 BINDIR	      = /usr/local/bin
 MANDIR	      = /usr/local/man/man1
 LIBDIR	      = /usr/local/share/yaze
+SYSCONFDIR    = /usr/local/etc
 
 # full speed or debugging to taste
 #OPTIMIZE      = -O2
@@ -44,7 +45,8 @@ INSTALL	      = install
 ###### you should not need to change anything below this line ######
 CWARN	      = -ansi -pedantic -Wall -Wshadow \
 		-Wpointer-arith -Wnested-externs -Winline
-CFLAGS        =	$(CWARN) $(OPTIMIZE) $(OPTIONS) -DLIBDIR=\"$(LIBDIR)/\"
+CFLAGS        =	$(CWARN) $(OPTIMIZE) $(OPTIONS) -DLIBDIR=\"$(LIBDIR)/\" \
+                -DSYSCONFDIR=\"$(SYSCONFDIR)\"
 
 SRCS	      = yaze.c simz80.c io.c bios.c monitor.c cdm.c
 DOC	      = README README-1.10 COPYING yaze.doc yaze.1 cdm.1 ChangeLog
diff --git a/bios.c b/bios.c
index 2864d62..d81ddc9 100644
--- a/bios.c
+++ b/bios.c
@@ -42,6 +42,10 @@ static char vcid[] __attribute__((unused)) = "$Id: bios.c,v 1.4 2004/04/23 09:50
 #endif /* lint */
 
 
+#ifndef SYSCONFDIR
+#define SYSCONFDIR "/usr/local/etc"
+#endif
+
 /* Z80 registers */
 #define AF	af[af_sel]
 #define BC	regs[regs_sel].bc
@@ -176,6 +180,10 @@ bios_init(const char *initfile)
 	    fn = buf;
 	    sprintf((char *) fn, "%s/%s", h, initfile);
 	}
+	if (access(fn, R_OK) != 0) {
+	    fn = buf;
+	    sprintf((char *) fn, "%s/%s", SYSCONFDIR, "yazerc");
+	}
     }
     if ((su = fopen(fn, "r")) == NULL)
 	return 0;
